src/sortStudents.ts

```diff
@@ -1,16 +1,80 @@
 
 export interface Student {
   // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
   // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
-
-export function sortStudents(students, sortBy, order) {
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+) : Student[] {
   // write your function
+  const person: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+      person.sort((a, b) => a.name.localeCompare(b.name));
+
+      break;
+
+    case SortType.Surname:
+      person.sort((a, b) => a.surname.localeCompare(b.surname));
+
+      break;
```

Comment:
Combine similar logic to one case and use computed keys
```suggestion
    case SortType.Name:
    case SortType.Surname:
      person.sort((a, b) => a[sortBy].localeCompare(b[sortBy]));

      break;
```